src/lib/components/CommunitySelectionModal/CommunitySelectionSearch.js

```diff
@@ -0,0 +1,168 @@
+// This file is part of React-Invenio-Deposit
+// Copyright (C) 2022 CERN.
+//
+// React-Invenio-Deposit is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from 'react';
+import {
+  EmptyResults,
+  Error,
+  InvenioSearchApi,
+  Pagination,
+  ReactSearchKit,
+  ResultsList,
+  ResultsLoader,
+  SearchBar,
+} from 'react-searchkit';
+import {
+  Container,
+  Grid,
+  Header,
+  Icon,
+  Item,
+  Menu,
+  Modal,
+  Segment,
+} from 'semantic-ui-react';
+import { CommunityListItem } from './CommunityListItem';
+import { OverridableContext } from 'react-overridable';
+import { i18next } from '@translations/i18next';
+import PropTypes from 'prop-types';
+
+const overriddenComponents = {
+  'communities.ResultsList.item': CommunityListItem,
+};
+
+export class CommunitySelectionSearch extends Component {
+  constructor(props) {
+    super(props);
+
+    this.apiEndpoints = {
```

Comment:
Question: Shouldn't this be in a more centralized place? (Maybe create an api.js under the lib folder, WDYT?)